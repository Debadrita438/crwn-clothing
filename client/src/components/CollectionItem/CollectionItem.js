import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cartAction';

import { 
    CollectionItemContainer, 
    CollectionItemImage, 
    CollectionFooterContainer, 
    NameStyles, 
    PriceStyles, 
    AddButton 
} from './CollectionItem.styles';

const mapDispatchToProps = dispatch => ({
    addItem: item => dispatch(addItem(item))
});

export const CollectionItem = ({ item, addItem }) => {
    const { name, price, imageUrl } = item;
    return (
        <CollectionItemContainer>
            <CollectionItemImage className='image' imageUrl={ imageUrl } />
            <CollectionFooterContainer>
                <NameStyles>{ name }</NameStyles>
                <PriceStyles>{ price }</PriceStyles>
            </CollectionFooterContainer>
            <AddButton onClick={ () => addItem(item) } inverted>
                Add to cart
            </AddButton>
        </CollectionItemContainer> 
    )
};
 
export default connect(null, mapDispatchToProps)(CollectionItem);