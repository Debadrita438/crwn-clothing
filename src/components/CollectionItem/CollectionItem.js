import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cartAction';

import { 
    CollectionItemContainer, 
    CollectionItemImage, 
    CollectionFooterContainer, 
    NameStyles, 
    PriceStyles, 
    AddButton 
} from './CollectionItem.styles';

const mapDispatchToProps = dispatch => ({
    addItem: item => dispatch(addItem(item))
});

const CollectionItem = ({ item, addItem }) => {
    const { name, price, imageUrl } = item;
    return (
        <CollectionItemContainer>
            <CollectionItemImage
                className='image'
                style={{
                    backgroundImage: `url(${imageUrl})`
                }}
            />
            <CollectionFooterContainer>
                <NameStyles>{ name }</NameStyles>
                <PriceStyles>{ price }</PriceStyles>
            </CollectionFooterContainer>
            <AddButton onClick={ () => addItem(item) } inverted>
                Add to cart
            </AddButton>
        </CollectionItemContainer> 
    )
};
 
export default connect(null, mapDispatchToProps)(CollectionItem);