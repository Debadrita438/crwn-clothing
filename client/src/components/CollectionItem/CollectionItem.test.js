import { shallow } from 'enzyme';

import { CollectionItem } from './CollectionItem';

describe('CollectionItem Component', () => {
    let wrapper;
    let mockAddItem;
    let mockImageUrl = 'www.testImage.com';
    let mockName = 'sneakers';
    let mockPrice = 10;
    beforeEach(() => {
        mockAddItem = jest.fn();

        const mockProps = {
            item: { 
                name: mockName,
                price: mockPrice,
                imageUrl: mockImageUrl
            },
            addItem: mockAddItem
        };

        wrapper = shallow(<CollectionItem { ...mockProps } />);
    });

    it('should render CollectionItem Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the image', () => {
        expect(wrapper.find('CollectionItemImage').prop('imageUrl')).toBe(mockImageUrl);
    });

    it('should render the name', () => {
        expect(wrapper.find('NameStyles').text()).toBe(mockName);
    });

    it('should render the price', () => {
        expect(parseInt(wrapper.find('PriceStyles').text())).toBe(mockPrice);
    });

    it('should add an item to cart when addItem is called', () => {
        wrapper.find('AddButton').simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
    });
});