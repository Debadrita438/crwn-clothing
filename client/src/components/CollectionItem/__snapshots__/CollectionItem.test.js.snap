// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem Component should render CollectionItem Component 1`] = `ShallowWrapper {}`;
